### ISSUE TITLE
:windo affects scrollview windows

### ISSUE THREAD

**Issue by jaekyeom** [dstein64/nvim-scrollview#45]
Hi,
I find this plugin very useful and mature.

One issue I'm having is that the `:windo` command applies to the scrollview windows as well.
(e.g., I cannot use `:windo diffthis`)

Can this issue be fixed?

**Comment by dstein64** [comment]
Hi @jaekyeom. Thanks for reporting the issue!

The documentation for windo says that it works like doing this:

```vim
CTRL-W t
:{cmd}
CTRL-W w
:{cmd}
etc.
```

That suggests that it would not operate in non-focusable windows, since `ctrl-w w` does not navigate to non-focusable windows. I've opened [Neovim Issue #15374](https://github.com/neovim/neovim/issues/15374).

I don't believe there's something that can be done from `nvim-scrollview` itself to prevent `windo` from executing commands in its non-focusable (scrollbar) windows.

A possible workaround could be to create a custom `Windo` command that only operates in focusable windows. For example:

```vim
command! -nargs=+ Windo
      \   for w in range(1, winnr('$'))
      \ |   if nvim_win_get_config(win_getid(w)).focusable
      \ |     execute w . 'wincmd w'
      \ |     execute <q-args>
      \ |   endif
      \ | endfor
```

Example usage:

```vim
:Windo echo winnr()
```

(unlike `:windo`, the example `:Windo` command above does not take an optional `[range]` argument)

**Comment by jaekyeom** [comment]
Wow, thanks for the explanation, workaround, and creating an issue!

**Comment by dstein64** [comment]
This is addressed in [Neovim #15378](https://github.com/neovim/neovim/pull/15378). The fix is included in [Neovim v0.6.0](https://github.com/neovim/neovim/releases/tag/v0.6.0).
